src/virtual_allocator.rs

```diff
@@ -148,6 +158,21 @@ impl VirtualMemoryManager {
         memory.free()?;
         Ok(())
     }
+
+    pub fn get_allocation_stats(&self) -> AllocationStats {
+        let (count, max) = self
+            .bitmap
```

Comment:
I don't think this is thread safe, because another thread could poke at `bitmap` while iteration is underway.
But current code isn't thread safe in general (at least `find_adjacent_free_blocks` isn't, maybe other parts aren't), so thread safety is out of scope imo.

Reply:
Indeed.

Reply:
Because the Allocator uses atomics (although they do not guarantee thread safety) i am keeping in line with that and also use atomic for the `maximum_tail_index` value and go through the vector of buckets when computing the stats, keeping the status quo.
IMHO the Allocator logic should be refactored but that is out of the scope of this PR.